Assets/Python/Stability.py

```diff
@@ -690,37 +690,35 @@ def calculateStability(iPlayer):
 	
 	for city in utils.getCityList(iPlayer):
 		iPopulation = city.getPopulation()
-		iModifier = 0
 		x = city.getX()
 		y = city.getY()
 		plot = gc.getMap().plot(x,y)
 		
 		bHistorical = (plot.getSettlerValue(iPlayer) >= 90)
 		
-		iOwnCulture = plot.getCulture(iPlayer)
-		iTotalCulture = 0
-		
-		bForeignCore = False
-		for iLoopPlayer in range(iNumPlayers):
-			iTempCulture = plot.getCulture(iLoopPlayer)
-			if plot.isCore(iLoopPlayer):
-				iTempCulture *= 2
-				if iLoopPlayer != iPlayer and iGameTurn > getTurnForYear(tBirth[iLoopPlayer]):
-					bForeignCore = True
-			iTotalCulture += iTempCulture
-			
-		if iTotalCulture != 0:
-			iCulturePercent = 100 * iOwnCulture / iTotalCulture
-		else:
-			iCulturePercent = 100
-				
-		bExpansionExceptions = ((bHistorical and iPlayer == iMongolia) or bTotalitarianism)
-		
 		# Expansion
 		if plot.isCore(iPlayer):
 			iCorePopulation += iCorePopulationModifier * iPopulation / 100
 			if bSingleCoreCity and iCurrentEra > iAncient: iCorePopulation += iCorePopulationModifier * iPopulation / 100
 		else:
+			iModifier = 0
+			
+			iOwnCulture = plot.getCulture(iPlayer)
+			iTotalCulture = 0
+			
+			for iLoopPlayer in range(iNumPlayers):
+				iTempCulture = plot.getCulture(iLoopPlayer)
+				if plot.isCore(iLoopPlayer):
+					iTempCulture *= 2
+				iTotalCulture += iTempCulture
+				
+			if iTotalCulture != 0:
+				iCulturePercent = 100 * iOwnCulture / iTotalCulture
+			else:
+				iCulturePercent = 100
+					
+			bExpansionExceptions = ((bHistorical and iPlayer == iMongolia) or bTotalitarianism)
+		

```

Comment:
This code is only required for non-core cities. I moved it so the calculations are skipped for core cities. (I did a check and the variables aren't used anywhere else in the code.

Reply:
Good idea. However I would still prefer to have the variable initialisations at the beginning of the loop. Otherwise there will be an exception if we ever decide to use them outside of the if clause. I'm talking about `iModifier = 0` and `iTotalCulture = 100` (the latter would also remove the need for that one else clause).